src/chain/tracker.rs

```
@@ -0,0 +1,303 @@
+                                        let now = SystemTime::now().duration_since(SystemTime::UNIX_EPOCH).unwrap().as_millis() as u64;
+
+                                        for (id, invoice) in &watched_accounts {
+                                            if events.iter().any(|event| was_balance_received_at_account(&invoice.address, &event.0.fields)) {
+                                                match invoice.check(&client, &watcher, &block).await {
+                                                    Ok(true) => {
+                                                        state.order_paid(id.clone()).await;
+                                                        id_remove_list.push(id.to_owned());
+                                                    }
+                                                    Ok(false) => (),
+                                                    Err(e) => {
+                                                        tracing::warn!("account fetch error: {0:?}", e);
+                                                    }
+                                                }
+                                            }
+
+                                            if invoice.death.0 >= now {
+                                                match invoice.check(&client, &watcher, &block).await {
+                                                    Ok(paid) => {
+                                                        if paid {
+                                                            state.order_paid(id.clone()).await;
+                                                        }
+
+                                                        id_remove_list.push(id.to_owned());
+                                                    }
+                                                    Err(e) => {
+                                                        tracing::warn!("account fetch error: {0:?}", e);
```

lots of noice about Paid being Null, need to ignore this error or preset Paid with value once invoice being created